main.tf

```diff
@@ -0,0 +1,135 @@
+/**
+ * Creates a Lambda function with associated role and policies that
+ * will run the ami-cleaner tool to get rid of AMIs that are no longer
+ * needed.
+ *
+ * Creates the following resources:
+ *
+ * * Lambda function
+ * * IAM role and policies to describe and delete AMIs and snapshots,
+ *   as well as write messages to Cloudwatch Logs
+ * * Cloudwatch Logs group
+ * * Cloudwatch Event to regularly run cleanup job.
+ *
+ * ## Usage
+ *
+ * ```hcl
+ * module "ami-clean-lambda" {
+ *   source            = "trussworks/lambda-amiclean/aws"
+ *   ami_clean_delete  = "true"
+ *   ami_clean_prefix  = "sample_app"
+ *   job_identifier    = "sample_app"
+ *   s3_bucket         = "sample_app_lambdas"
+ *   version_to_deploy = "2.7"
+ * }
+ * ```
+ *
+ * For more details on the capabilities of the ami-cleaner tool, as well
+ * as how to deploy it, see <https://github.com/trussworks/truss-aws-tools>.
+ */
+
+locals {
+  pkg  = "truss-aws-tools"
+  name = "ami-cleaner"
+}
+
+data "aws_region" "current" {}
+
+# This is the main policy this job is going to need.
+data "aws_iam_policy_document" "main" {
+  # Allow describing AMIs.
+  statement {
+    sid    = "DescribeImages"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeImages",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow describing EBS snapshots.
+  statement {
+    sid    = "DescribeEBSSnapshots"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeSnapshots",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow deregistering AMIs.
+  statement {
+    sid    = "DeregisterImage"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DeregisterImage",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow deleteing EBS snapshots.
+  statement {
+    sid    = "DeleteSnapshot"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DeleteSnapshot",
+    ]
+
+    resources = ["arn:aws:ec2:${data.aws_region.current.name}::snapshot/*"]
+  }
+}
+
+# Create the policy for the document we just added above.
+resource "aws_iam_policy" "main" {
+  name   = "${local.name}-${var.job_identifier}-policy"
+  policy = "${data.aws_iam_policy_document.main.json}"
+}
+
+# Lambda function
+module "amiclean_lambda" {
+  source = "github.com/trussworks/terraform-aws-lambda"
```

Comment:
I will *definitely* do that before this gets published to the registry -- right now, I just want to get these into github so that I can make sure it works with some testing, and then I'll update both this and the terraform-aws-lambda module and tag them with a version.